job-server/src/spark.jobserver/JobServer.scala

```diff
@@ -66,6 +75,44 @@ object JobServer {
 
   }
 
+  private def storeInitialJars(config: Config, jarManager: ActorRef): Unit = {
+    if(config.hasPath("spark.jobserver.job-jar-uris")) {
+      val initialJarsConfig = config.getConfig("spark.jobserver.job-jar-uris").root
+
+      logger.info("Adding initial job jars: {}", initialJarsConfig.render())
+
+      val initialJars =
+        initialJarsConfig
+          .asScala
+          .map { case (key, value) => (key, value.unwrapped.toString) }
+          .toMap
+
+      // Ensure that the jars exist
+      for(jarPath <- initialJars.values) {
+        val f = new java.io.File(jarPath)
+        if(!f.exists) {
+          val msg =
+            if(f.isAbsolute) {
+              s"Inital Jar File $jarPath does not exist"
+            } else {
+              s"Inital Jar File $jarPath (${f.getAbsolutePath}) does not exist"
+            }
+
+          throw new java.io.IOException(msg)
```

Comment:
If there is an error with the initial JARs, then SJS won't start, and will inform you about the failing JAR. That behavior was by design, do you think another behavior would be better?